llarp/iwp/session.cpp

```diff
@@ -596,6 +596,14 @@ namespace llarp
         }
       }
       SendMACK();
+      if(m_EncryptNext && !m_EncryptNext->empty())
+      {
+        auto self = shared_from_this();
+        m_Parent->QueueWork([self, data = std::move(m_EncryptNext)] {
```

Comment:
You can drop the `auto self = ...` and just initialize it in the lambda captures:
```C++
m_Parent->QueueWork([self = shared_from_this(), data = std::move(m_EncryptNext)] {
```
which should save one shared pointer reference increment/decrement.